pkg/virt-handler/vm.go

```diff
@@ -291,6 +292,12 @@ func (d *VMHandlerDispatch) processVmUpdate(vm *v1.VM, shouldDeleteVm bool) erro
 		return err
 	}
 
+	// Map Container Registry Disks to block devices Libvirt can consume
+	vm, err = registrydisk.MapRegistryDisks(vm)
+	if err != nil {
+		return err
+	}
```

Comment:
I'm not debating whether the docker image feature makes sense - just the approach to implementation. There are countless different ways to do dynamic storage and this docker image is just one possibility. Adding code into the core to support every different approach does not make sense - these should be done as pluggable storage technologies via a generic framework to enable arbitrary third party implementations, of which this docker container approach is just one such impl.